storage: add option force_mask= and permissions_xattr= (#744)
containers/storage: docs/containers-storage.conf.5.md

**ignore_chown_errors** = "false"
  ignore_chown_errors can be set to allow a non privileged user running with a  single UID within a user namespace to run containers. The user can pull and use any image even those with multiple uids.  Note multiple UIDs will be squashed down to the default uid in the container.  These images will have no separation between the users in the container. (default: false)

**force_mask** = "0000"

>   force_mask sets a octal permissions mask and those bits are set for each new file or directory created in the storage.  The final permission for the new files and directories is set to ORIGINAL_PERMISSION|FORCE_MASK. (default: 0000)

Reviewer: ```suggestion
  force_mask sets an octal permission mask and those bits are set for each new file or directory created in the storage.  The final permission for the new files and directories is set to ORIGINAL_PERMISSION|FORCE_MASK. (default: 0000)
```

**permissions_xattr** = ""

>   it specifies an extended attribute where the original permissions for the file are stored in the format UID:GID:MODE.

Reviewer: ```suggestion
  This option specifies an extended attribute where the original permissions for the file are stored in the format UID:GID:MODE.
```

**ignore_chown_errors** = "false"
  ignore_chown_errors can be set to allow a non privileged user running with a  single UID within a user namespace to run containers. The user can pull and use any image even those with multiple uids.  Note multiple UIDs will be squashed down to the default uid in the container.  These images will have no separation between the users in the container. (default: false)

> **force_mask** = "0000|native|shared|private"

Reviewer 2: Need to mention here that it requires mount_program.

Reviewer 2: Update man page, then LGTM

Author: fixed.  Should we mark it as experimental for now so we keep some freedom to change it if we need?

Reviewer 2: SGTM

**ignore_chown_errors** = "false"
  ignore_chown_errors can be set to allow a non privileged user running with a  single UID within a user namespace to run containers. The user can pull and use any image even those with multiple uids.  Note multiple UIDs will be squashed down to the default uid in the container.  These images will have no separation between the users in the container. (default: false)

**force_mask** = "0000|native|shared|private"
  force_mask sets an octal permission mask and those bits are set for each new file or directory created in the storage.  The final permission for the new files and directories is set to ORIGINAL_PERMISSION|FORCE_MASK.  The values "native", "shared", "private" are also accepted.  (default: "native")
  "native": it is equivalent to 0000.  The files maintain their original permission mask.
  "private": it is equivalent to 0700.  The owner has rwx access to the files.
  "shared": it is equivalent to 0755.  The owner has rwx access to the files and everyone else can read, access and execute them.

>   It is possible to use **force_mask** only when **mount_program** is specified.  It is an experimental feature.

Reviewer 2: How about `Note: The force_mask Flag is an experimental feature, it could change in the future.`

**ignore_chown_errors** = "false"
  ignore_chown_errors can be set to allow a non privileged user running with a  single UID within a user namespace to run containers. The user can pull and use any image even those with multiple uids.  Note multiple UIDs will be squashed down to the default uid in the container.  These images will have no separation between the users in the container. (default: false)

**force_mask** = "0000|native|shared|private"
  force_mask sets an octal permission mask and those bits are set for each new file or directory created in the storage.  The final permission for the new files and directories is set to ORIGINAL_PERMISSION|FORCE_MASK.  The values "native", "shared", "private" are also accepted.  (default: "native")
  "native": it is equivalent to 0000.  The files maintain their original permission mask.
  "private": it is equivalent to 0700.  The owner has rwx access to the files.
  "shared": it is equivalent to 0755.  The owner has rwx access to the files and everyone else can read, access and execute them.

> **permissions_xattr** = ""

Reviewer 2: Should this be a constant?  Or at least document the one that fuse-overlay will use.  Not sure why users would expect more then one xattr?

Reviewer 2: I think this is useless complexity.  Just store the attributes in storage.dac.permissions or something and be done with it.

Author: I didn't want to hard code "fuse-overlayfs" in containers/storage.  I can document it better that it is expected to be used with fuse-overlayfs, and perhaps explain how fuse-overlayfs uses it

Reviewer 2: Why not just define a "storage" xattr and have fuse-overlyfs deal with it?

Author: fuse-overlayfs currently handles every `user.fuseoverlayfs*` as a private namespace, while any other xattr is not filtered out.  If we add a `storage` xattr we'd have to handle a separate namespace and not permit it inside of the containers, so `user.storage` or `user.containers` won't be available to container processes.
